src/Framework/Framework/Resources/Scripts/metadata/typeMap.ts

```diff
@@ -3,18 +3,27 @@ import { keys } from "../utils/objects";
 
 let types: TypeMap = {};
 
-export function getTypeInfo(typeId: string): TypeMetadata {
-    var result = types[typeId];
-    if (!result) {
-        throw new Error(`Cannot find type metadata for '${typeId}'!`);
+export function getTypeInfo(typeId: string | object): TypeMetadata {
+
+    if (typeof typeId === "string") {
+        var result = types[typeId];
+        if (result) {
+            return result;
+        }
+    }
+    else if (typeof typeId === "object") {
+        var typeInfo = typeId as any;
+        if (ko.unwrap(typeInfo?.type) === "dynamic")
+            return { type: "dynamic" };
```

Comment:
```suggestion
            return { type: "dynamic", properties: {} };
```

I think this could eliminate number of conditions bellow. AFAIK are supposed to treat any additional properties as dynamic (to not fall apart on inheritance), so dynamic type is essentially a type without any information about its properties.

Reply:
I agree but it causes problems in some other places. We can refactor it a little if we want in another PR